fn/io.go

```diff
@@ -0,0 +1,51 @@
+package fn
+
+import (
+	"os"
+)
+
+// FileRemovalOnErr is a flag used to indicate the removal or not of a file on a
+// failure.
+type FileRemovalOnErr bool
```

Comment:
What is the use case of this? Or put it this way, in `lnd`, do we have more `KeepFileOnErr` or `RemoveFileOnErr`? Asking because `WriteFile` is a fundamental method, and wanna to make sure the API is easy and efficient to use.

For instance, if most of the time we just wanna remove the file, I think it can be the default behavior of `WriteFile`, and creating a new method `WriteFileNoRemoval` - this way it can save some engineering time since it abstracts more (no need to pass a third param, etc.).

Reply:
I will check how many file removals on error are present in the code to define the default functionality, and then make the other into a separate function, as you suggested.

PD: this PR is linked to [PR #6768](https://github.com/lightningnetwork/lnd/pull/6768) the current function signature was based on [this thread](https://github.com/lightningnetwork/lnd/pull/6768/files/ec837c063cad01281f0f699b6c4c74b7474f4b76#r930138186).

Reply:
I've submitted an updated version with the WriteFile and WriteFileRemove functions.

I’ve defined the 'no-remove' version as the default because I tracked how many times a file was removed on error after an os.WriteFile call. The remove version is used less frequently:
- 15 times with no removal
- 7 times with removal